packages/donor/src/screens/bookDonation/BookDonationScreen.tsx

```diff
@@ -17,13 +17,16 @@ import NoAppointments from "../../assets/images/NO Appointments.svg";
 import { Player } from "@lottiefiles/react-lottie-player";
 import DropAnimation from "../../assets/animations/drop.json";
 import { DonationSlotToBook } from "../../state/AppointmentToBookStore";
+import Popup, { PopupProps } from "../../components/basic/Popup";
+import WarningLogo from "../../assets/images/warning.svg";
 
 export interface BookDonationScreenProps {
   availableAppointments: AvailableAppointment[];
   isFetching: boolean;
   firstName?: string;
   onSlotSelected: (donationSlot: DonationSlotToBook) => void;
   defaultHospital: Hospital | "";
+  onSlotSelectedPopUpProps: Pick<PopupProps, "open" | "onBack" | "onApproved">;
```

Comment:
Why did  you decide to use `Pick<PopupProps, "open" | "onBack" | "onApproved">`?

Personally, having to write a function name in a string is a smell to me.

I suggest you just add 3 explicit props to this screen: `tooClosePopupOpen`, `onApproveCloseAppointment` and `onDeclineCloseAppointment`.

Reply:
I uses it because :

1. To show that there is a strong connection to the popup props
2. Avoid duplication
3. Avoid load on p the props

I don't like the way that pick uses strings too, but using explicit props is a problem if this page will have another popup on the future.